on darwin:

```
builder for '/nix/store/kly7k84mzclcaffvwpyfhw7mgb1i11fh-libcyaml-1.1.0.drv' failed with exit code 2; last 10 log lines:
  clang -I include -DVERSION_MAJOR=1 -DVERSION_MINOR=1 -DVERSION_PATCH=0 -DVERSION_DEVEL=0  -std=c11 -Wall -Wextra -pedantic -Wconversion -O2 -DNDEBUG -fPIC  -c -o build/release/shared/src/mem.o src/mem.c
  clang -I include -DVERSION_MAJOR=1 -DVERSION_MINOR=1 -DVERSION_PATCH=0 -DVERSION_DEVEL=0  -std=c11 -Wall -Wextra -pedantic -Wconversion -O2 -DNDEBUG -fPIC  -c -o build/release/shared/src/free.o src/free.c
  clang -I include -DVERSION_MAJOR=1 -DVERSION_MINOR=1 -DVERSION_PATCH=0 -DVERSION_DEVEL=0  -std=c11 -Wall -Wextra -pedantic -Wconversion -O2 -DNDEBUG -fPIC  -c -o build/release/shared/src/load.o src/load.c
  clang -I include -DVERSION_MAJOR=1 -DVERSION_MINOR=1 -DVERSION_PATCH=0 -DVERSION_DEVEL=0  -std=c11 -Wall -Wextra -pedantic -Wconversion -O2 -DNDEBUG -fPIC  -c -o build/release/shared/src/save.o src/save.c
  clang -I include -DVERSION_MAJOR=1 -DVERSION_MINOR=1 -DVERSION_PATCH=0 -DVERSION_DEVEL=0  -std=c11 -Wall -Wextra -pedantic -Wconversion -O2 -DNDEBUG -fPIC  -c -o build/release/shared/src/util.o src/util.c
  clang -I include -DVERSION_MAJOR=1 -DVERSION_MINOR=1 -DVERSION_PATCH=0 -DVERSION_DEVEL=0  -std=c11 -Wall -Wextra -pedantic -Wconversion -O2 -DNDEBUG -fPIC  -c -o build/release/shared/src/utf8.o src/utf8.c
  clang -o build/release/libcyaml.so.1 build/release/shared/src/mem.o build/release/shared/src/free.o build/release/shared/src/load.o build/release/shared/src/save.o build/release/shared/src/util.o build/release/shared/src/utf8.o -lyaml  -Wl,-soname=libcyaml.so.1 -shared
  ld: unknown option: -soname=libcyaml.so.1
  clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
  make: *** [Makefile:130: build/release/libcyaml.so.1] Error 1
```

--- a/pkgs/development/libraries/libcyaml/default.nix
+++ b/pkgs/development/libraries/libcyaml/default.nix
@@ -1,0 +1,26 @@
+{ stdenv, lib, fetchFromGitHub
+, libyaml
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libcyaml";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "tlsa";
+    repo = "libcyaml";
+    rev = "v${version}";
+    sha256 = "0428p0rwq71nhh5nzcbapsbrjxa0x5l6h6ns32nxv7j624f0zd93";
+  };
+
+  buildInputs = [ libyaml ];
+
+  makeFlags = [ "VARIANT=release" "PREFIX=$(out)" ];
+
+  meta = with lib; {
+    homepage = "https://github.com/tlsa/libcyaml";
+    description = "C library for reading and writing YAML";
+    license = licenses.isc;
+    platforms = platforms.linux;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -14341,6 +14341,8 @@ in
 
   libyamlcpp = callPackage ../development/libraries/libyaml-cpp { };
 
+  libcyaml = callPackage ../development/libraries/libcyaml { };
+
   rang = callPackage ../development/libraries/rang { };
 
   libyamlcpp_0_3 = pkgs.libyamlcpp.overrideAttrs (oldAttrs: {

